packages/server/lib/plugins/child/run_require_async_child.js

```diff
@@ -118,7 +118,7 @@ function run (ipc, file, projectRoot) {
 
       return (await bundleRequire({ filepath: file })).mod
     } catch (err) {
-      if (err.stack.includes(`Cannot find package 'esbuild'`)) {
+      if (err.stack.includes(`Cannot find module 'esbuild'`)) {
```

Comment:
```suggestion
      if (err.code === 'MODULE_NOT_FOUND') {
```

Reply:
Actually, I think we should just be try/catch'ing the `require.resolve` and then if that fails do the early return

Reply:
Like this: https://github.com/cypress-io/cypress/pull/21576/files#diff-d25a897afab55c6be61f5c5064524f60036f4ef0e6b399ebb7f0c851ea0a22b7

Reply:
No, that doesn't make sense. MODULE_NOT_FOUND is a legit error that can come from any module that bundleRequire fails to find when it tries to load the config file

Reply:
While using the error code looks nice, that’s a legit error we conditionally want to throw. In your proposed change we also SWAP preferring node’s import over bundleRequire, which is not something we want to do. This OSS a change in functionality. bundleRequire shims esm-only libraries like `findUp` Which is why we prefer it. 

This PR is a typo fix. We’re code frozen. The code flow is well-tested and I don’t want to change it.

Reply:
@JessicaSachs Did you look at the code thing I linked, it does not use the `err.code` and doesn't change the control flow, it's:

```ts
try {
  require.resolve('esbuild')
} catch {
  debug(`User doesn't have esbuild. Going to use native node imports.`)

  // We cannot replace the initial `require` with `await import` because
  // Certain modules cannot be dynamically imported
  return await import(file)
}

debug(`They have esbuild, so we'll load the configFile via bundleRequire`)
const { bundleRequire } = require('bundle-require')

return (await bundleRequire({ filepath: file })).mod
```

Which is just being more explicit about what we're guarding for failure, doesn't rely on string matching, which might be different based on node version or change at some point in the future, and doesn't change the current control flow. 

Regardless, we should make sure we can get a failing test in here with whatever the change ends up being.

Reply:
The only functional issue I have with the above code is that you no longer throw the errors from `bundleRequire`, which may include TS compilation errors, etc.

**On the stability of the current code:**
The `err.stack` string hasn't changed in a long time -- it's safe to assert on across node versions and I think you yourself suggested I make this change ~5 weeks ago. The reason it's different is because I made a typo initially.

**On why this wasn't caught in our infrastructure:**
Because we install a lot of packages automatically to make our system-tests run faster, `esbuild` was always in the `require.resolve` chain as a transitive dependency and we never hit this code path in the tests.

So yes, agreed, we need to fix our system test infra. I've been talking with @flotwig about WHY the tests I wrote didn't fail.

Reply:
Ok, @JessicaSachs and I met on this - it's fine, let's just land this change for now to get it working